jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/DynamicUtils.java

```diff
@@ -275,7 +276,7 @@ private static long executeSubProcess(StatefulKnowledgeSessionImpl ksession,
                 List<String> businessKeys = new ArrayList<>();
                 businessKeys.add(((WorkflowProcessInstanceImpl) processInstance).getCorrelationKey());
                 businessKeys.add(processId);
-                businessKeys.add(String.valueOf(System.currentTimeMillis()));
+                businessKeys.add(UUID.randomUUID().toString());
```

Comment:
So I guess issue was in case there were 2 subprocesses created in a succession, for example with multi-instance/loop subprocess, right? And at least 2 of them had the same timestamp... That's really quick!

Reply:
Correct. Two parallel subprocesses from same parent created in the same millisecond... specially now having H2 pure in memory did the trick.